components/mime.rst

```diff
@@ -0,0 +1,541 @@
+        ->text('Lorem ipsum...')
+        ->html('<p>Lorem ipsum...</p>')
+
+        // contents obtained from a PHP resource
+        ->text(fopen('/path/to/emails/user_signup.txt', 'r'))
+        ->html(fopen('/path/to/emails/user_signup.html', 'r'))
+    ;
+
+.. tip::
+
+    You can also use Twig templates to render the HTML and text contents. Read
+    the :ref:`mime-component-twig-integration` section later in this article to
+    learn more.
+
+Embedding Images
+----------------
+
+If you want to display images inside your email contents, you must embed them
+instead of adding them as attachments. First, use the ``embed()`` or
+``embedFromPath()`` method to add an image from a file or resource::
+
+    $email = (new Email())
+        // ...
+        // get the image contents from a PHP resource
+        ->embed(fopen('/path/to/images/logo.png', 'r'), 'logo')
+        // get the image contents from an existing file
+        ->embedFromPath('/path/to/images/signature.gif', 'footer-signature')
+    ;
+
+The second optional argument of both methods is the image name ("Content-ID" in
+the MIME standard). Its value is an arbitrary string used later to reference the
+images inside the HTML contents::
+
+    $email = (new Email())
+        // ...
+        ->embed(fopen('/path/to/images/logo.png', 'r'), 'logo')
+        ->embedFromPath('/path/to/images/signature.gif', 'footer-signature')
+        // reference images using the syntax 'cid:' + "image embed name"
+        ->html('<img src="cid:logo" /> ... <img src="cid:footer-signature" /> ...')
+    ;
+
+File Attachments
+----------------
+
+Use the ``attachFromPath()`` method to attach files that exist in your file system::
+
+    $email = (new Email())
+        // ...
+        ->attachFromPath('/path/to/documents/terms-of-use.pdf')
+        // optionally you can tell email clients to display a custom name for the file
+        ->attachFromPath('/path/to/documents/privacy.pdf', 'Privacy Policy')
+        // optionally you can provide an explicit MIME type (otherwise it's guessed)
+        ->attachFromPath('/path/to/documents/contract.doc', 'Contract', 'application/msword')
+    ;
+
+Alternatively you can use the ``attach()`` method to attach contents generated
+with PHP resources::
+
+    $email = (new Email())
+        // ...
+        ->attach(fopen('/path/to/documents/contract.doc', 'r'))
+    ;
+
+Creating Raw Email Messages
+---------------------------
+
+This is useful for advanced applications that need absolute control over every
+email part. It's not recommended for applications with regular email
+requirements because it adds a considerable complexity for no real gain.
+
+Before continuing, it's important to have a look at the low level structure of
+an email message. Consider a message which includes some content as both text
+and HTML, a single PNG image included in those contents and a PDF file attached
+to it. The MIME standard structures that message as the following tree:
```

Comment:
As a matter of fact, you can describe this email with many different structures, but this one is the only one that works on the main/major/popular email clients.